dlt/sources/helpers/transform.py

```diff
@@ -24,3 +26,31 @@ def _filter(_: TDataItem) -> bool:
         return count > max_items
 
     return _filter
+
+
+def pivot(columns: Union[str, List[str]], prefix: str) -> ItemTransformFunctionNoMeta[TDataItem]:
+    """Pivot a list of columns into a dictionary.
+
+    Args:
+        columns (Union[str, List[str]]): list of column names
```

Comment:
pls check out https://github.com/dlt-hub/dlt/issues/1002 the proposed solution. the arguments are
- json path to the element(s) containing lists of lists, let's put default "$" to select the root. also we have jsonpath_ng in our deps so please use it (jsonpath.py in common)
- a prefix for column prefix. so columns start with prefix and have an index as suffix. we can also accept a list of columns like the above. but then you should make sure that len of the list is exactly the number of columns